packages/ide-extension/src/telemetry/types.ts

```diff
@@ -0,0 +1,39 @@
+import type { Disposable } from 'vscode';
+import type { TelemetryClient } from 'applicationinsights';
+
+export interface TelemetryReporter extends Disposable {
+    client: TelemetryClient;
+    commonProperties?: TelemetryCommonProperties;
+    enabled: boolean;
+}
+
+export interface TelemetryEventProperties {
+    readonly [key: string]: string;
+}
+
+export type TelemetryEvent = TelemetryUIEvent;
+
+interface TelemetryBaseEvent {
+    name: string;
+    properties?: TelemetryEventProperties;
+}
+
+export interface TelemetryCommonProperties extends TelemetryEventProperties {
+    'cmn.appstudio': 'true' | 'false';
+    'cmn.devspace': string;
+    'common.os': string;
+    'common.nodeArch': string;
+    'common.platformversion': string;
+    'common.extname': string;
+    'common.extversion': string;
+}
+
+export interface TelemetryUIEvent extends TelemetryBaseEvent {
+    name: 'USER_INTERACTION';
+    properties: TelemetryUIEventProps;
+}
+
+export interface TelemetryUIEventProps extends TelemetryEventProperties {
+    action: string;
+    [prop: string]: string;
+}
```

Comment:
These general event definitions look fine. 
Now we need to add the two concrete interfaces for telemetry event properties that are logged. From the of the code this looks like

```typescript
export interface TelemetryUIOpenTutorialProps {
    action: 'OPEN_TUTORIAL';
    title: string;
    primaryTag: string;
}

export interface TelemetryUIOpenBlogProps {
    action: 'OPEN_BLOG';
    title: string;
    primaryTag: string;
}
```

The links are a bit confusing, because in this case the redux action is triggered for telemetry reasons only and the properties of redux action e.g. `KHUB_OPEN_BLOGS` matches exactly the properties of telemetry event `USER_INTERACTION` with property action `OPEN_BLOG`.
This is not always the case, sometimes the redux action has different properties then the telemetry event. To ensure that we have documented all telemetry events with properties, they should be defined here and used as return value of the `action-map.ts` mappings.